bootloaders/riotboot/Makefile

```diff
@@ -8,7 +8,7 @@ BOARD ?= samr21-xpro
 FEATURES_REQUIRED += riotboot
 
 # Disable unused modules
-CFLAGS += -DNDEBUG -DLOG_LEVEL=LOG_NONE
+CFLAGS += -DNDEBUG -DLOG_LEVEL=LOG_NONE -DBOOTLOADER
```

Comment:
I would keep the term `BOOTLOADER` or use `RIOT_BOOTLOADER` instead of `RIOTBOOT`. `riotboot` is associated with a module that any application can use so I think we should try and keep it separate from stuff only the bootloader needs.

Reply:
I don't care. I think that `RIOT_BOOTLOADER` reduces chances on a conflicting define.

I was planning to split this commit from this PR, so I'll change it over there!

Reply:
> I don't care. I think that `RIOT_BOOTLOADER` reduces chances on a conflicting define.

I do a little. Please keep BOOTLOADER over RIOT_BOOTLOADER if RIOTBOOT doesn't resonate.

Reply:
> How can people not be of the opinion that "RIOTBOOT" is the right name for this define? wink

I would think something that isn't `RIOTBOOT` is better since this isn't necessarily a riotboot boot-loader issue, but for any bootloader, e.g: for a case where e.g. `mcuboot` or similar would be supported, I imagine this initialization would need to be skipped. But that could be changed if and when the need arises.

> Well, "RIOT_BOOTLOADER" is just an arbitrary but new name of RIOT's bootloader, "riotboot". Why a new name?

Ok, you are right.